token-plugins-guide.md

```diff
@@ -0,0 +1,180 @@
+# 1inch Token Plugins: A Comprehensive Guide for Extending ERC20 Functionalities
+
+[Overview](#overview)
+
+[Primary Benefits](#primary-benefits)
+
+[Implementation](#implementation)
+
+[Overview of Contract Structure](#overview-of-contract-structure)
+   - [ERC20Plugins.sol (Abstract Plugin-enabled ERC20 contract)](#erc20pluginssol-plugin-enabled-erc20-token-contract)
+   - [Plugin.sol (Abstract Plugin contract)](#pluginsol-abstract-plugin-contract)
+
+[Deployed Examples](#deployed-examples)
+
+[Helpful Links](#helpful-links)
+
+## Overview
+
+Token plugins are smart contracts that extend the capabilities of plugin-enabled ERC20 tokens and plugin-enabled ERC20 wrappers. They allow for additional accounting to be added, without the need to deposit your tokens into a contract. **_This is a revolutionary step in the evolution of DeFi_**, as the entire ecosystem has historically relied upon the transfer of tokens in and out of external contracts to achieve enhanced capabilities! 
+
+By design, token plugins prevent several common attack vectors and inherently create a fundamental level of security within every token incentive mechanism. Users no longer have to trust their funds in an external contract, whether it be for farming, borrowing/lending, or delegating, etc., therefore adding a fundamental layer of security.
+
+## Primary Benefits
+- 100% permissionless, open to all participants.
+- Risk-free participation: Token plugins can only update with an account’s balance change. They cannot approve, mint, burn, transfer, or overwrite storage.
+- Users can connect multiple plugins, allowing for simultaneous participation in multiple incentive programs or governance systems, etc.
+- Implementation is less than 100 lines of code; very simple to adopt.
```

Comment:
Not sure, maybe 150? Need to check with `sloc` software.

Reply:
Nice estimate :)

sloc showing:
- ERC20Plugins.sol: 133 lines
- Plugins.sol: 17 lines